### PR TITLE
Ansible Playbook Bundle tests implementation

### PR DESCRIPTION
**Pull request by jakubschwan**
PR with re-based branch master/dev-apb to master branch. Included compilation fixes and also fixes for maven test execution.
Also fixed versions in pom.xml files.
Branch master/dev-apb is without the changes for master.
Commit "Resolve compilation issues after branch rebase to master" 25fbaa8 was not reviewed on master/dev-apb branch.

### CODE REVIEW

File: framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/deployment/DatabaseDeployment.java [hunk 1]
```diff
@@ -27,7 +28,7 @@
      *
      * @return Database URL.
      */
-    URL getUrl();
+    Optional<URL> getUrl();
```
Comment (sutaakar) [inline framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/deployment/DatabaseDeployment.java]: What is the reason for this change? Is database URL used anywhere?

File: framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/constants/OpenShiftApbConstants.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    // RH-SSO
+    public static final String SSO_URL = "apb_sso_url";
+    public static final String SSO_REALM = "apb_sso_realm";
+    public static final String BUSINESS_CENTRAL_SSO_CLIENT = "apb_sso_businesscentral_client";
+    public static final String BUSINESS_CENTRAL_SSO_SECRET = "apb_sso_businesscentral_client_secret";
+    public static final String KIE_SERVER_SSO_CLIENT = "apb_kieserver_sso_client";
+    public static final String KIE_SERVER_SSO_SECRET = "apb_kieserver_sso_client_secret";
+    //public static final String SSO_CLIENT = "apb_sso_client";
+    //public static final String SSO_CLIENT_SECRET = "apb_sso_client_secret";
```
Comment (sutaakar) [inline framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/constants/OpenShiftApbConstants.java]: Please delete commented code.

File: framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioApb.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    }
+
+    @Override
+    public List<ControllerDeployment> getControllerDeployments() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public SsoDeployment getSsoDeployment() {
+        throw new UnsupportedOperationException("Not supported yet.");
```
Comment (sutaakar) [inline framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioApb.java]: Is there a reason why SSO deployment is not returned? It seems that it can be configured for this scenario.

File: framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/builder/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioBuilderApb.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+public class ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioBuilderApb implements ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioBuilder {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioBuilderApb.class);
+    private final Map<String, String> extraVars = new HashMap<>();
+    
+    private boolean deploySSO = false;
+
+    public ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioBuilderApb() {
+        // Required values to create persitence values.
```
Comment (sutaakar) [inline framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/builder/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioBuilderApb.java]:
```suggestion
        // Required values to create persistence values.
```

File: framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/builder/WorkbenchKieServerScenarioBuilderApb.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        return new WorkbenchKieServerScenarioApb(extraVars);
+    }
+
+    @Override
+    public WorkbenchKieServerScenarioBuilder withExternalMavenRepo(String repoUrl, String repoUserName, String repoPassword) {
+        extraVars.put(OpenShiftApbConstants.MAVEN_REPO_URL, repoUrl);
+        extraVars.put(OpenShiftApbConstants.MAVEN_REPO_USER, repoUserName);
+        extraVars.put(OpenShiftApbConstants.MAVEN_REPO_PWD, repoPassword);
+        return this;
+        //throw new UnsupportedOperationException("Not supported for APB.");
```
Comment (sutaakar) [inline framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/builder/WorkbenchKieServerScenarioBuilderApb.java]: Delete this line?

File: framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ProjectImpl.java [hunk 6]
```diff
@@ -98,6 +102,68 @@ public void processTemplateAndCreateResources(URL templateUrl, Map<String, Strin
...
+                logger.info(podName);
+                File outputDirectory = new File(System.getProperty(APB_BUILDS_LOGS_OUTPUT_DIRECTORY, DEFAULT_APB_BUILDS_LOG_OUTPUT_DIRECTORY));
+                if (!outputDirectory.isDirectory()) {
+                    outputDirectory.mkdir();
+                }
+                File logFile = new File(outputDirectory, projectName + "." + podName + LOG_SUFFIX);
+                FileUtils.copyInputStreamToFile(istream, logFile);
+            } ,args.toArray(new String[args.size()]));
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Failed acquire from java.util.concurrent.Semaphore", e);
```
Comment (sutaakar) [inline framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ProjectImpl.java]: Please adjust InterruptedException handling in similar way as above.

File: test-cloud/pom.xml [hunk 7]
```diff
@@ -236,9 +257,6 @@
     </profile>
     <profile>
       <id>openshift</id>
-      <properties>
```
Comment (sutaakar) [inline test-cloud/pom.xml]: What is the reason to move these properties configurations to submodules?
It just causes duplication from my point of view.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/ClusteredWorkbenchKieServerPersistentScenarioLdapIntegrationTest.java [hunk 8]
```diff
@@ -52,10 +53,14 @@ public static void initializeDeployment() {
                 .withLdapRoleRecursion(LdapSettingsConstants.ROLE_RECURSION)
                 .withLdapDefaultRole(LdapSettingsConstants.DEFAULT_ROLE).build();
 
+        try {
         deploymentScenario = deploymentScenarioFactory.getClusteredWorkbenchKieServerDatabasePersistentScenarioBuilder()
                 .withLdapSettings(ldapSettings).withExternalMavenRepo(MavenConstants.getMavenRepoUrl(),
                         MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
                 .build();
+        } catch (UnsupportedOperationException ex) {
+            Assume.assumeFalse(ex.getMessage().startsWith("Not supported"));
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/ClusteredWorkbenchKieServerPersistentScenarioLdapIntegrationTest.java]: What is the purpose of this catch?

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioLdapIntegrationTest.java [hunk 9]
```diff
@@ -23,17 +23,19 @@
...
 import org.kie.cloud.integrationtests.testproviders.HttpsWorkbenchTestProvider;
 import org.kie.cloud.integrationtests.testproviders.OptaplannerTestProvider;
 import org.kie.cloud.integrationtests.testproviders.ProcessTestProvider;
 import org.kie.cloud.integrationtests.util.LdapSettingsConstants;
-import org.kie.cloud.tests.common.ScenarioDeployer;
 import org.kie.cloud.maven.constants.MavenConstants;
+import org.kie.cloud.tests.common.AbstractCloudIntegrationTest;
+import org.kie.cloud.tests.common.ScenarioDeployer;
 
+@Category(ApbNotSupported.class) // https://issues.jboss.org/browse/KIECLOUD-101
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioLdapIntegrationTest.java]: The issue seems to be resolved, does the test work now?

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/WorkbenchKieServerPersistentScenarioLdapIntegrationTest.java [hunk 10]
```diff
@@ -35,10 +35,11 @@
 import org.kie.cloud.integrationtests.testproviders.ProcessTestProvider;
 import org.kie.cloud.integrationtests.testproviders.ProjectBuilderTestProvider;
 import org.kie.cloud.integrationtests.util.LdapSettingsConstants;
-import org.kie.cloud.tests.common.ScenarioDeployer;
 import org.kie.cloud.maven.constants.MavenConstants;
+import org.kie.cloud.tests.common.AbstractCloudIntegrationTest;
+import org.kie.cloud.tests.common.ScenarioDeployer;
 
-@Category(Baseline.class)
+@Category({Baseline.class,ApbNotSupported.class}) // https://issues.jboss.org/browse/KIECLOUD-101
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/WorkbenchKieServerPersistentScenarioLdapIntegrationTest.java]: Same as above

File: test-cloud/pom.xml (hunk 7 above)
Comment (jakubschwan) [reply test-cloud/pom.xml]: SkipTests property from this module was ignored and all tests were skipped.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/ClusteredWorkbenchKieServerPersistentScenarioLdapIntegrationTest.java (hunk 8 above)
Comment (jakubschwan) [reply test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/ldap/ClusteredWorkbenchKieServerPersistentScenarioLdapIntegrationTest.java]: If scenario configuration is not possible, then all tests are skipped.

File: framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioApb.java (hunk 3 above)
Comment (jakubschwan) [reply framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioApb.java]: It's not possible to configure SSO for this scenario. We are not able to create SSO client for more then 1 set of Kie Servers.

File: framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/deployment/DatabaseDeployment.java (hunk 1 above)
Comment (jakubschwan) [reply framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/deployment/DatabaseDeployment.java]: To return URL in same type as for other deployments. If URL for DB is not needed, we can remove this method.
